# ITMGT 25.03 EE
# Assignment 3B

This is a data-cleaning assignment and is the second of two parts. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.

In addition, you have a new file called **studentprofiles.csv** which has first and last names plus the genders of the students.

The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called **cleanrecweekdata3b.csv** **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats3b.html**. The requirements for the summary statistics are described below. **(10 points)**

The outputs are similar to that of Assignment 3A, except that you will now use Pandas instead of Python lists.

This entire assignment (3B) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Monday, July 9, 2019** at 12:00NN.

Note: I understand you all need to review for LT 2. Think of this Assignment as practice. It's not meant to take you as long as 3A.

#### Bonus Points:
* +2 if submitted over (or before) the weekend (July 13-14)
* +2 if CSS and div tags are used (but please don't spend too much time on this because you still need to review for the Long Test)

In [63]:
## Code goes here

import pandas as pd

df = pd.read_csv("recweekdata.csv") # read
df2 = pd.read_csv("studentprofiles.csv") # read

In [64]:
df.head() # just looking

firstname   lastname birthmonth  birthday  birthyear      course  \
0     name1  lastname1         11        26       1998       BS LM   
1     name2  lastname2        May        26       2001  Management   
2     name3  lastname3          5        16       1997      BS CTM   
3     name4  lastname4        Dec         9         97      BS ITE   
4     name5  lastname5        Sep         5         97      BS CTM   

        phonenumber  
0  +63 918 302 6210  
1      927 212 6276  
2  +63 918 253 7017  
3      918 714 4360  
4  +63 927 462 1674

In [65]:
df2.head() # just looking

firstname   lastname  gender
0     name1  lastname1  Female
1     name2  lastname2  Female
2     name3  lastname3  Female
3     name4  lastname4    Male
4     name5  lastname5  Female

In [66]:
## birthmonth

bmDict = {"1":"January","2":"February","3":"March","4":"April","5":"May","6":"June",
        "7":"July","8":"August","9":"September","10":"October","11":"November","12":"December",
        "Jan":"January","Feb":"February","Mar":"March","Apr":"April","Jun":"June",
        "Jul":"July","Aug":"August","Sep":"September","Oct":"October","Nov":"November",
        "Dec":"December","January":"January","February":"February","March":"March","April":"April",
        "May":"May","June":"June","July":"July","August":"August","September":"September","October":"October",
        "November":"November","December":"December"} # dictionary to clean

df.loc[:,"birthmonth"] = df.loc[:,"birthmonth"].apply(lambda month: bmDict[month]) # subject all to Dict

df.head()

firstname   lastname birthmonth  birthday  birthyear      course  \
0     name1  lastname1   November        26       1998       BS LM   
1     name2  lastname2        May        26       2001  Management   
2     name3  lastname3        May        16       1997      BS CTM   
3     name4  lastname4   December         9         97      BS ITE   
4     name5  lastname5  September         5         97      BS CTM   

        phonenumber  
0  +63 918 302 6210  
1      927 212 6276  
2  +63 918 253 7017  
3      918 714 4360  
4  +63 927 462 1674

In [67]:
## birthyear

import re

def yearCleaner(year):
    if year >= 1900 and year < 2000: # 19XX
        return re.sub("^19","",str(year)) # remove 19
    elif year >= 2000: # 20XX
        return re.sub("^20","",str(year)) # remove 20
    else:
        return str(year) # for XX, keep as is

df.loc[:,"birthyear"] = df.loc[:,"birthyear"].apply(yearCleaner) # apply to all

df.head()

firstname   lastname birthmonth  birthday birthyear      course  \
0     name1  lastname1   November        26        98       BS LM   
1     name2  lastname2        May        26        01  Management   
2     name3  lastname3        May        16        97      BS CTM   
3     name4  lastname4   December         9        97      BS ITE   
4     name5  lastname5  September         5        97      BS CTM   

        phonenumber  
0  +63 918 302 6210  
1      927 212 6276  
2  +63 918 253 7017  
3      918 714 4360  
4  +63 927 462 1674

In [68]:
## view all possible responses for course

df.loc[:,["firstname","course"]].groupby(["course"]).count()

firstname
course                                            
BS CTM                                         186
BS ITE                                         177
BS LM                                          192
BS ME                                          199
BS MGT                                         216
CTM                                            208
Communications Technology                      195
ITE                                            212
Information Technology Entrepreneurship        201
LM                                             194
Legal Management                               199
ME                                             213
MGT                                            199
Management                                     214
Management Engineering                         195

In [69]:
## course

courseDict = {"BS CTM":"BS Communications Technology Management", "BS ITE":"BS Information Technology Entrepreneurship",
             "BS LM":"BS Legal Management", "BS ME":"BS Management Engineering", "BS MGT":"BS Management",
             "CTM":"BS Communications Technology Management", "ITE":"BS Information Technology Entrepreneurship",
             "LM":"BS Legal Management", "ME":"BS Management Engineering", "MGT":"BS Management",
             "Communications Technology":"BS Communications Technology Management",
             "Information Technology Entrepreneurship":"BS Information Technology Entrepreneurship",
             "Legal Management":"BS Legal Management", "Management Engineering":"BS Management Engineering",
             "Management":"BS Management"} # dictionary to clean

df.loc[:,"course"] = df.loc[:,"course"].apply(lambda course: courseDict[course]) # subject all to Dict

df

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26        98   
1        name2     lastname2        May        26        01   
2        name3     lastname3        May        16        97   
3        name4     lastname4   December         9        97   
4        name5     lastname5  September         5        97   
5        name6     lastname6      March         2        99   
6        name7     lastname7   November        10        98   
7        name8     lastname8     August        28        99   
8        name9     lastname9       July        10        97   
9       name10    lastname10  September        14        01   
10      name11    lastname11   December         8        98   
11      name12    lastname12        May         4        97   
12      name13    lastname13    January        28        99   
13      name14    lastname14    January        11        99   
14      name15    lastname15       June         6        98   
15      name16    lastname16      April        20        99   
16      name17    lastname17    January         3        97   
17      name18    lastname18      April         8        97   
18      name19    lastname19   December        28        97   
19      name20    lastname20      March        10        97   
20      name21    lastname21       July        13        99   
21      name22    lastname22  September        15        99   
22      name23    lastname23      April        15        97   
23      name24    lastname24    October         1        97   
24      name25    lastname25        May         6        97   
25      name26    lastname26   December         5        99   
26      name27    lastname27      March         4        97   
27      name28    lastname28   November        10        00   
28      name29    lastname29    October        10        99   
29      name30    lastname30        May        28        99   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25        97   
2971  name2972  lastname2972        May        14        98   
2972  name2973  lastname2973    October         9        99   
2973  name2974  lastname2974     August        12        99   
2974  name2975  lastname2975   November        18        97   
2975  name2976  lastname2976       July        24        98   
2976  name2977  lastname2977        May        26        98   
2977  name2978  lastname2978    October        26        99   
2978  name2979  lastname2979   February        13        97   
2979  name2980  lastname2980       July         3        99   
2980  name2981  lastname2981   February        17        98   
2981  name2982  lastname2982      March        28        00   
2982  name2983  lastname2983   December         3        98   
2983  name2984  lastname2984   November        13        00   
2984  name2985  lastname2985   February        23        99   
2985  name2986  lastname2986      March         2        99   
2986  name2987  lastname2987    October         6        97   
2987  name2988  lastname2988   February        20        97   
2988  name2989  lastname2989    January        17        99   
2989  name2990  lastname2990       July        14        97   
2990  name2991  lastname2991       July        27        00   
2991  name2992  lastname2992      March        26        99   
2992  name2993  lastname2993      April         1        00   
2993  name2994  lastname2994      April        10        99   
2994  name2995  lastname2995        May        21        97   
2995  name2996  lastname2996      March        28        97   
2996  name2997  lastname2997        May        19        98   
2997  name2998  lastname2998       July         3        98   
2998  name2999  lastname2999   December         2        98   
2999  name3000  lastname3000       July        13        97   

                                          course       phonenumber  
0                            

In [70]:
## phonenumber (part 1)

def numberCleaner(number):
    if number[0] == "9": # 9XX ...
        return re.sub("^9","+63 9",number) # add +63(space)
    else:
        return number

df.loc[:,"phonenumber"] = df.loc[:,"phonenumber"].apply(numberCleaner) # apply to all

df

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26        98   
1        name2     lastname2        May        26        01   
2        name3     lastname3        May        16        97   
3        name4     lastname4   December         9        97   
4        name5     lastname5  September         5        97   
5        name6     lastname6      March         2        99   
6        name7     lastname7   November        10        98   
7        name8     lastname8     August        28        99   
8        name9     lastname9       July        10        97   
9       name10    lastname10  September        14        01   
10      name11    lastname11   December         8        98   
11      name12    lastname12        May         4        97   
12      name13    lastname13    January        28        99   
13      name14    lastname14    January        11        99   
14      name15    lastname15       June         6        98   
15      name16    lastname16      April        20        99   
16      name17    lastname17    January         3        97   
17      name18    lastname18      April         8        97   
18      name19    lastname19   December        28        97   
19      name20    lastname20      March        10        97   
20      name21    lastname21       July        13        99   
21      name22    lastname22  September        15        99   
22      name23    lastname23      April        15        97   
23      name24    lastname24    October         1        97   
24      name25    lastname25        May         6        97   
25      name26    lastname26   December         5        99   
26      name27    lastname27      March         4        97   
27      name28    lastname28   November        10        00   
28      name29    lastname29    October        10        99   
29      name30    lastname30        May        28        99   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25        97   
2971  name2972  lastname2972        May        14        98   
2972  name2973  lastname2973    October         9        99   
2973  name2974  lastname2974     August        12        99   
2974  name2975  lastname2975   November        18        97   
2975  name2976  lastname2976       July        24        98   
2976  name2977  lastname2977        May        26        98   
2977  name2978  lastname2978    October        26        99   
2978  name2979  lastname2979   February        13        97   
2979  name2980  lastname2980       July         3        99   
2980  name2981  lastname2981   February        17        98   
2981  name2982  lastname2982      March        28        00   
2982  name2983  lastname2983   December         3        98   
2983  name2984  lastname2984   November        13        00   
2984  name2985  lastname2985   February        23        99   
2985  name2986  lastname2986      March         2        99   
2986  name2987  lastname2987    October         6        97   
2987  name2988  lastname2988   February        20        97   
2988  name2989  lastname2989    January        17        99   
2989  name2990  lastname2990       July        14        97   
2990  name2991  lastname2991       July        27        00   
2991  name2992  lastname2992      March        26        99   
2992  name2993  lastname2993      April         1        00   
2993  name2994  lastname2994      April        10        99   
2994  name2995  lastname2995        May        21        97   
2995  name2996  lastname2996      March        28        97   
2996  name2997  lastname2997        May        19        98   
2997  name2998  lastname2998       July         3        98   
2998  name2999  lastname2999   December         2        98   
2999  name3000  lastname3000       July        13        97   

                                          course       phonenumber  
0                            

In [71]:
## phonenumber (part 2)

def spaceRemover(number):
    return re.sub("\s","",number) # change space to no space

df.loc[:,"phonenumber"] = df.loc[:,"phonenumber"].apply(spaceRemover) # apply to all

In [72]:
df # just checking

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26        98   
1        name2     lastname2        May        26        01   
2        name3     lastname3        May        16        97   
3        name4     lastname4   December         9        97   
4        name5     lastname5  September         5        97   
5        name6     lastname6      March         2        99   
6        name7     lastname7   November        10        98   
7        name8     lastname8     August        28        99   
8        name9     lastname9       July        10        97   
9       name10    lastname10  September        14        01   
10      name11    lastname11   December         8        98   
11      name12    lastname12        May         4        97   
12      name13    lastname13    January        28        99   
13      name14    lastname14    January        11        99   
14      name15    lastname15       June         6        98   
15      name16    lastname16      April        20        99   
16      name17    lastname17    January         3        97   
17      name18    lastname18      April         8        97   
18      name19    lastname19   December        28        97   
19      name20    lastname20      March        10        97   
20      name21    lastname21       July        13        99   
21      name22    lastname22  September        15        99   
22      name23    lastname23      April        15        97   
23      name24    lastname24    October         1        97   
24      name25    lastname25        May         6        97   
25      name26    lastname26   December         5        99   
26      name27    lastname27      March         4        97   
27      name28    lastname28   November        10        00   
28      name29    lastname29    October        10        99   
29      name30    lastname30        May        28        99   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25        97   
2971  name2972  lastname2972        May        14        98   
2972  name2973  lastname2973    October         9        99   
2973  name2974  lastname2974     August        12        99   
2974  name2975  lastname2975   November        18        97   
2975  name2976  lastname2976       July        24        98   
2976  name2977  lastname2977        May        26        98   
2977  name2978  lastname2978    October        26        99   
2978  name2979  lastname2979   February        13        97   
2979  name2980  lastname2980       July         3        99   
2980  name2981  lastname2981   February        17        98   
2981  name2982  lastname2982      March        28        00   
2982  name2983  lastname2983   December         3        98   
2983  name2984  lastname2984   November        13        00   
2984  name2985  lastname2985   February        23        99   
2985  name2986  lastname2986      March         2        99   
2986  name2987  lastname2987    October         6        97   
2987  name2988  lastname2988   February        20        97   
2988  name2989  lastname2989    January        17        99   
2989  name2990  lastname2990       July        14        97   
2990  name2991  lastname2991       July        27        00   
2991  name2992  lastname2992      March        26        99   
2992  name2993  lastname2993      April         1        00   
2993  name2994  lastname2994      April        10        99   
2994  name2995  lastname2995        May        21        97   
2995  name2996  lastname2996      March        28        97   
2996  name2997  lastname2997        May        19        98   
2997  name2998  lastname2998       July         3        98   
2998  name2999  lastname2999   December         2        98   
2999  name3000  lastname3000       July        13        97   

                                          course    phonenumber  
0                            BS 

In [73]:
# carrier

carrierDict = {"+63917":"Globe","+63927":"Globe","+63918":"Smart"}

df["carrier"] = df.loc[:,"phonenumber"].str[0:6].apply(lambda number: carrierDict[number])
df

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26        98   
1        name2     lastname2        May        26        01   
2        name3     lastname3        May        16        97   
3        name4     lastname4   December         9        97   
4        name5     lastname5  September         5        97   
5        name6     lastname6      March         2        99   
6        name7     lastname7   November        10        98   
7        name8     lastname8     August        28        99   
8        name9     lastname9       July        10        97   
9       name10    lastname10  September        14        01   
10      name11    lastname11   December         8        98   
11      name12    lastname12        May         4        97   
12      name13    lastname13    January        28        99   
13      name14    lastname14    January        11        99   
14      name15    lastname15       June         6        98   
15      name16    lastname16      April        20        99   
16      name17    lastname17    January         3        97   
17      name18    lastname18      April         8        97   
18      name19    lastname19   December        28        97   
19      name20    lastname20      March        10        97   
20      name21    lastname21       July        13        99   
21      name22    lastname22  September        15        99   
22      name23    lastname23      April        15        97   
23      name24    lastname24    October         1        97   
24      name25    lastname25        May         6        97   
25      name26    lastname26   December         5        99   
26      name27    lastname27      March         4        97   
27      name28    lastname28   November        10        00   
28      name29    lastname29    October        10        99   
29      name30    lastname30        May        28        99   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25        97   
2971  name2972  lastname2972        May        14        98   
2972  name2973  lastname2973    October         9        99   
2973  name2974  lastname2974     August        12        99   
2974  name2975  lastname2975   November        18        97   
2975  name2976  lastname2976       July        24        98   
2976  name2977  lastname2977        May        26        98   
2977  name2978  lastname2978    October        26        99   
2978  name2979  lastname2979   February        13        97   
2979  name2980  lastname2980       July         3        99   
2980  name2981  lastname2981   February        17        98   
2981  name2982  lastname2982      March        28        00   
2982  name2983  lastname2983   December         3        98   
2983  name2984  lastname2984   November        13        00   
2984  name2985  lastname2985   February        23        99   
2985  name2986  lastname2986      March         2        99   
2986  name2987  lastname2987    October         6        97   
2987  name2988  lastname2988   February        20        97   
2988  name2989  lastname2989    January        17        99   
2989  name2990  lastname2990       July        14        97   
2990  name2991  lastname2991       July        27        00   
2991  name2992  lastname2992      March        26        99   
2992  name2993  lastname2993      April         1        00   
2993  name2994  lastname2994      April        10        99   
2994  name2995  lastname2995        May        21        97   
2995  name2996  lastname2996      March        28        97   
2996  name2997  lastname2997        May        19        98   
2997  name2998  lastname2998       July         3        98   
2998  name2999  lastname2999   December         2        98   
2999  name3000  lastname3000       July        13        97   

                                          course    phonenumber carrier  
0                       

In [74]:
df3 = pd.merge(df, df2.loc[:,["firstname","gender"]], on = "firstname") 
# merge basis firstname
# do not include lastname from df2

In [75]:
df3 # just checking

firstname      lastname birthmonth  birthday birthyear  \
0        name1     lastname1   November        26        98   
1        name2     lastname2        May        26        01   
2        name3     lastname3        May        16        97   
3        name4     lastname4   December         9        97   
4        name5     lastname5  September         5        97   
5        name6     lastname6      March         2        99   
6        name7     lastname7   November        10        98   
7        name8     lastname8     August        28        99   
8        name9     lastname9       July        10        97   
9       name10    lastname10  September        14        01   
10      name11    lastname11   December         8        98   
11      name12    lastname12        May         4        97   
12      name13    lastname13    January        28        99   
13      name14    lastname14    January        11        99   
14      name15    lastname15       June         6        98   
15      name16    lastname16      April        20        99   
16      name17    lastname17    January         3        97   
17      name18    lastname18      April         8        97   
18      name19    lastname19   December        28        97   
19      name20    lastname20      March        10        97   
20      name21    lastname21       July        13        99   
21      name22    lastname22  September        15        99   
22      name23    lastname23      April        15        97   
23      name24    lastname24    October         1        97   
24      name25    lastname25        May         6        97   
25      name26    lastname26   December         5        99   
26      name27    lastname27      March         4        97   
27      name28    lastname28   November        10        00   
28      name29    lastname29    October        10        99   
29      name30    lastname30        May        28        99   
...        ...           ...        ...       ...       ...   
2970  name2971  lastname2971   February        25        97   
2971  name2972  lastname2972        May        14        98   
2972  name2973  lastname2973    October         9        99   
2973  name2974  lastname2974     August        12        99   
2974  name2975  lastname2975   November        18        97   
2975  name2976  lastname2976       July        24        98   
2976  name2977  lastname2977        May        26        98   
2977  name2978  lastname2978    October        26        99   
2978  name2979  lastname2979   February        13        97   
2979  name2980  lastname2980       July         3        99   
2980  name2981  lastname2981   February        17        98   
2981  name2982  lastname2982      March        28        00   
2982  name2983  lastname2983   December         3        98   
2983  name2984  lastname2984   November        13        00   
2984  name2985  lastname2985   February        23        99   
2985  name2986  lastname2986      March         2        99   
2986  name2987  lastname2987    October         6        97   
2987  name2988  lastname2988   February        20        97   
2988  name2989  lastname2989    January        17        99   
2989  name2990  lastname2990       July        14        97   
2990  name2991  lastname2991       July        27        00   
2991  name2992  lastname2992      March        26        99   
2992  name2993  lastname2993      April         1        00   
2993  name2994  lastname2994      April        10        99   
2994  name2995  lastname2995        May        21        97   
2995  name2996  lastname2996      March        28        97   
2996  name2997  lastname2997        May        19        98   
2997  name2998  lastname2998       July         3        98   
2998  name2999  lastname2999   December         2        98   
2999  name3000  lastname3000       July        13        97   

                                          course    phonenumber carrier  \
0                      

In [76]:
df3.to_csv("cleanrecweekdata3b.csv", index=False) # export to csv

# Required Summary Statistics

#### These were the questions from 3A:

1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

#### Additional Questions:
* How many Globe subscribers are: Male? Female?
* How many Smart subscribers are: Male? Female?

Answer these questions in an html file called **stats3b.html**. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
             <tr><td>Male Globe subscribers:</td><td>{}</td></tr>
             <tr><td>Female Globe subscribers:</td><td>{}</td></tr>
             <tr><td>Male Smart subscribers:</td><td>{}</td></tr>
             <tr><td>Female Smart subscribers:</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [77]:
# birthmonth

dfbm = df3.loc[:,["firstname","birthmonth"]].groupby(["birthmonth"]).count().reset_index()
dfbm

birthmonth  firstname
0       April        267
1      August        241
2    December        276
3    February        232
4     January        270
5        July        249
6        June        259
7       March        246
8         May        273
9    November        277
10    October        178
11  September        232

In [78]:
# rearrange the months

dfbm["monthnumber"] = [4,8,12,2,1,7,6,3,5,11,10,9]
dfbm = dfbm.sort_values(by=['monthnumber'],ascending=True).copy()

dfbm = dfbm.reset_index() # new index

In [79]:
dfbm = dfbm.drop(columns=['index']).copy() # delete old index

In [80]:
dfbm = dfbm.drop(columns=['monthnumber']).copy() # delete monthnumber
dfbm

birthmonth  firstname
0     January        270
1    February        232
2       March        246
3       April        267
4         May        273
5        June        259
6        July        249
7      August        241
8   September        232
9     October        178
10   November        277
11   December        276

In [81]:
# birthday

dfbd = df3.loc[:,["firstname","birthday"]].groupby(["birthday"]).count().reset_index()
dfbd

birthday  firstname
0          1        112
1          2        106
2          3         90
3          4        110
4          5        100
5          6         99
6          7         88
7          8        111
8          9         98
9         10        111
10        11        105
11        12         98
12        13        119
13        14        102
14        15        121
15        16        115
16        17        116
17        18        107
18        19        113
19        20         91
20        21        123
21        22        103
22        23        116
23        24        108
24        25        116
25        26        116
26        27        102
27        28        104

In [82]:
# birthyear

dfby = df3.loc[:,["firstname","birthyear"]].groupby(["birthyear"]).count().reset_index()
dfby

birthyear  firstname
0        00        380
1        01        362
2        97        756
3        98        746
4        99        756

In [83]:
# rearrange the years

dfby["yearorder"] = [4,5,1,2,3]
dfby = dfby.sort_values(by=['yearorder'],ascending=True).copy()

dfby = dfby.reset_index() # new index
dfby = dfby.drop(columns=['index']).copy() # delete old index
dfby = dfby.drop(columns=['yearorder']).copy() # delete yearorder
dfby

birthyear  firstname
0        97        756
1        98        746
2        99        756
3        00        380
4        01        362

In [84]:
# course

dfcou = df3.loc[:,["firstname","course"]].groupby(["course"]).count().reset_index()
dfcou

course  firstname
0     BS Communications Technology Management        589
1  BS Information Technology Entrepreneurship        590
2                         BS Legal Management        585
3                               BS Management        629
4                   BS Management Engineering        607

In [85]:
# carrier

dfcar = df3.loc[:,["firstname","carrier"]].groupby(["carrier"]).count().reset_index()
dfcar

carrier  firstname
0   Globe       2045
1   Smart        955

In [86]:
# How many Globe subscribers are: Male? Female?
# How many Smart subscribers are: Male? Female?

dfcar2 = df3.loc[:,["firstname","carrier","gender"]].groupby(["carrier","gender"]).count().reset_index()
dfcar2

carrier  gender  firstname
0   Globe  Female        984
1   Globe    Male       1061
2   Smart  Female        494
3   Smart    Male        461

In [87]:
# html format
html = '''
<link href="div.css" rel="stylesheet">

<div class="Table">
    <div class="Title">
        <p>Summary Statistics</p>
    </div>
    <div class="Heading">
        <div class="Cell">
            <p>Stat</p>
        </div>
        <div class="Cell">
            <p>Value</p>
        </div>
    </div>
    <div class="Row">
        <div class="Cell">
            <p>People born in each month</p>
        </div>
        <div class="Cell">
            {}
        </div>
    </div>
    <div class="Row">
        <div class="Cell">
            <p>People born on each day</p>
        </div>
        <div class="Cell">
            {}
        </div>
    </div>
    <div class="Row">
        <div class="Cell">
            <p>People born in each year</p>
        </div>
        <div class="Cell">
            {}
        </div>
    </div>
    <div class='Row'>
        <div class='Cell'>
            <p>People coming from each course</p>
        </div>
        <div class='Cell'>
            {}
        </div>
    </div>
    <div class='Row'>
        <div class='Cell'>
            <p>People having numbers from Globe and Smart</p>
        </div>
        <div class='Cell'>
            {}
        </div>
    </div>
    <div class='Row'>
        <div class='Cell'>
            <p>Male Globe subscribers:</p>
        </div>
        <div class='Cell'>
            {}
        </div>
    </div>
    <div class='Row'>
        <div class='Cell'>
            <p>Female Globe subscribers:</p>
        </div>
        <div class='Cell'>
            {}
        </div>
    </div>
    <div class='Row'>
        <div class='Cell'>
            <p>Male Smart subscribers:</p>
        </div>
        <div class='Cell'>
            {}
        </div>
    </div>
    <div class='Row'>
        <div class='Cell'>
            <p>Female Globe subscribers:</p>
        </div>
        <div class='Cell'>
            {}
        </div>
    </div>
 </div>
'''

In [105]:
# generate lines for birthmonth

statMonth = dfbm.loc[:,["birthmonth",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1]), axis=1)
statMonth = ''.join(statMonth)

In [106]:
# generate lines for birthday

statDay = dfbd.loc[:,["birthday",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1]), axis=1)
statDay = ''.join(statDay)

In [107]:
# generate lines for birthyear

statYear = dfby.loc[:,["birthyear",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1]), axis=1)
statYear = ''.join(statYear)

In [108]:
# generate lines for course

statCou = dfcou.loc[:,["course",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1]), axis=1)
statCou = ''.join(statCou)

In [109]:
# generate lines for carrier

statCar = dfcar.loc[:,["carrier",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1]), axis=1)
statCar = ''.join(statCar)

In [110]:
# generate lines for Male Globe

statCarMG = dfcar2.loc[1:1,["carrier","gender",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1],x[2]), axis=1)
statCarMG = ''.join(statCarMG)

# generate lines for Female Globe

statCarFG = dfcar2.loc[0:0,["carrier","gender",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1],x[2]), axis=1)
statCarFG = ''.join(statCarFG)

# generate lines for Male Smart

statCarMS = dfcar2.loc[3:3,["carrier","gender",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1],x[2]), axis=1)
statCarMS = ''.join(statCarMS)

# generate lines for Female Smart

statCarFS = dfcar2.loc[2:2,["carrier","gender",
            "firstname"]].apply(lambda x: '<div class="Table"><div class="Row"><div class="Cell">{}</div><div class="Cell">{}</div><div class="Cell">{}</div></div></div>'.format(x[0],x[1],x[2]), axis=1)
statCarFS = ''.join(statCarFS)

In [111]:
# put together
htmlDone = html.format(statMonth,statDay,statYear,statCou,statCar,statCarMG,statCarFG,statCarMS,statCarFS)

In [112]:
# see it on Python

from IPython.core.display import HTML
HTML(htmlDone)

In [113]:
# create a file

with open("stats3b.html","w+") as htmlFile:
    htmlFile.write(htmlDone)